mlir/lib/Pass/CMakeLists.txt

```diff
@@ -16,4 +16,6 @@ add_mlir_library(MLIRPass
   LINK_LIBS PUBLIC
   MLIRAnalysis
   MLIRIR
+  PRIVATE
+  ${LLVM_ATOMIC_LIB}
```

Comment:
Why isn't this part of `add_mlir_library`? Seems a bit random to add this here and there right now.

Reply:
fair point
I just put it in the specific places where it was needed (ie using the libatomic symbols) the same way it was done in other places in llvm (mlir never had support for this case).
but this will require some 'manual' maintenance as it is not obvious where these functions are used.
adding it globally to the helper might be a better avenue.